ampd/src/solana/verifier_set_verifier.rs

```diff
@@ -40,213 +43,90 @@ pub fn parse_gateway_event(tx: &EncodedConfirmedTransactionWithStatusMeta) -> Re
-        );
-        return Vote::FailedOnChain;
+    let axelar_verifier_set_hash = hash_verifier_set(&verifier_set_conf.verifier_set);
+    if &axelar_verifier_set_hash == new_signers_hash {
+        return Vote::SucceededOnChain;
     }
+    Vote::FailedOnChain
+}
 
-    if verifier_set_threshold != solana_quorum {
-        error!(
-            tx_id,
-            axelar_threshold = verifier_set_threshold,
-            solana_quorum,
-            "Verifier set threshold do not match."
-        );
-        return Vote::FailedOnChain;
-    }
+fn hash_verifier_set(verifier_set: &VerifierSet) -> [u8; 32] {
+    let mut hasher = Keccak256::new();
 
-    for (solana_addr, solana_weight) in solana_signers.iter().zip(solana_weights.iter()) {
-        let solana_addr_hex = solana_addr.encode_hex::<String>();
-        let Some((addr, signer)) = verifier_set.signers.get_key_value(&solana_addr_hex) else {
-            error!(
-                tx_id,
-                solana_addr_hex, "Lookup for solana signer address failed on axelar verifier set",
-            );
-            return Vote::FailedOnChain;
-        };
-        let signer_address = signer.address.to_string();
-        if *addr != signer_address {
-            error!(
-                tx_id,
-                verifier_set_map_key=addr,
-                verifier_set_inner_signer_address=signer_address,
-                "Axelar verifier set has inconsistencies. Map key (Address) is different than the inner signer address.",
-            );
-            return Vote::FailedOnChain;
-        }
-        let signer_pub_key = match &signer.pub_key {
-            PublicKey::Ecdsa(hb) | PublicKey::Ed25519(hb) => hb,
-        };
-        if solana_addr.as_ref() != signer_pub_key.as_slice() {
-            let signer_pub_key_hex = signer.pub_key.encode_hex::<String>();
-            error!(
-                tx_id,
-                solana_addr_hex,
-                signer_pub_key_hex,
-                "Solana address is different than Axelar signer public key.",
-            );
-            return Vote::FailedOnChain;
-        }
+    // Length prefix the bytes to be hashed to prevent hash collisions
+    hasher.update(verifier_set.signers.len().to_be_bytes());
 
-        if *solana_weight != signer.weight.u128() {
-            error!(
-                tx_id,
-                solana_addr_hex,
-                solana_weight,
-                axelar_signer_weight = signer.weight.u128(),
-                "Signer weight differs",
-            );
-            return Vote::FailedOnChain;
+    verifier_set.signers.values().for_each(|signer| {
+        match signer.pub_key {
+            PublicKey::Ecdsa(_) => hasher.update(b"secp256k1"),
+            PublicKey::Ed25519(_) => hasher.update(b"ed25519"),
         }
-    }
-    Vote::SucceededOnChain
+        hasher.update(signer.pub_key.as_ref());
+        hasher.update(signer.weight.to_le_bytes());
+    });
```

Comment:
Need to take into account this comment: https://github.com/eigerco/solana-axelar-internal/issues/410#issuecomment-2333791886 . Should we add that for completeness ? (maybe is redudant since it would be computed from the pubkey)

Reply:
hmm I think that it's redundant. The only thing it might protect against is if the same signer is operating on a different chain that is not `axelar` and it is in the same verifier sets on both chains. I think that will never happen.